src/concuerror.erl

```diff
@@ -40,8 +45,10 @@ start(Options, LogMsgs) ->
   Estimator = concuerror_estimator:start_link(Options),
   LoggerOptions = [{estimator, Estimator},{processes, Processes}|Options],
   Logger = concuerror_logger:start(LoggerOptions),
+  ProcessGenerator = concuerror_callback:start_process_generator(Options),
```

Comment:
Two things here:
- I suggest using a separate module for `process_generator` instead of placing it into `concuerror_callback`. It could even be a `gen_server`.
- The `concuerror_estimator` is started here because both the scheduler and the logger talk to it. In your current version the `process_generator` only communicates with the scheduler (and instrumented processes) so it could be spawned in an initialisation phase of the callback module.